Run integration tests against `nightly` (not `latest`)

avelanarius opened the pull request:
The `latest` tag of `quesma/quesma` is the latest released stable release of Quesma, while `nightly` is the newest build of Quesma. Integration tests used to run against `quesma/quesma:latest` and not the last `nightly` build of Quesma, which meant that we wouldn't catch some failures before release.

Fix this issue by running the integration tests against `nightly` and always building the newest `quesma/quesma:nightly` in the `integration-test.yml` GitHub workflow.

avelanarius commented on .github/workflows/integration-tests.yml:
```diff
@@ -10,6 +10,42 @@ on:
         description: 'Commit hash to run the tests'
         required: true
 jobs:
+  build-quesma-docker-image:
+    strategy:
+      matrix:
+        module: [ "quesma" ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with: ## @TODO REMOVE
+          ref: ${{ github.event.inputs.GIT_REF }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          cache-dependency-path: ${{ matrix.module }}/go.sum
+          go-version: '1.22'
+
+      - name: Build and export
+        uses: docker/build-push-action@v6
+        with:
+          context: ${{ matrix.module }}/.
+          tags: ${{ matrix.module }}:latest
+          outputs: type=docker,dest=/tmp/image.tar
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+        env:
+          DOCKER_BUILD_SUMMARY: false
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.module }}
+          path: /tmp/image.tar
+          retention-days: 1
```
Blatant copy-paste from another workflow. This copy-paste will get fixed by: https://github.com/QuesmaOrg/quesma/pull/862

mieciu replied:
maybe let's merge #862 and apply it then ?

avelanarius replied:
👍

avelanarius replied:
Fixed in the latest version

avelanarius commented:
In the amended commit I removed the duplication of Docker building step and fixed a couple small bugs.

Tested the integration test workflow here: https://github.com/avelanarius/quesma/actions/workflows/integration-tests.yml
(run 5 is a run on latest master, run 4 is a run on a branch with deliberate panic and the test caught that - used correct new image, ignore runs 1-3)
